File: frontend/src/app/core/helpers/signalr/SignalRFlowMessageResponse.ts
import { FlowMessageAction } from "../../entities/flow-messages/flow-message/enums/FlowMessageAction";
import { FlowMessageEventType } from "../../entities/flow-messages/flow-message/enums/FlowMessageEventType";
import { FlowMessageType } from "../../entities/flow-messages/flow-message/enums/FlowMessageType";
import { SignalRFlowMessageButtonResponse } from "./SignalRFlowMessageButtonResponse";
import { SignalRFlowMessageListItemResponse } from "./SignalRFlowMessageListItemResponse";

export interface SignalRFlowMessageResponse {
  content: string;
  type: FlowMessageType;
  action: FlowMessageAction;
  eventType: FlowMessageEventType;
  buttons: SignalRFlowMessageButtonResponse[];
  buttonListDisplayText: string;
  listItems: SignalRFlowMessageListItemResponse[];
}

File: backend/SprintBusiness/SprintBuisness.Contracts/Whatsapp/Dtos/MessageResponse/SignalRFlowMessageResponse.cs
using SprintBusiness.Domain.Flows.FlowMessages.Enums;

namespace SprintBuisness.Contracts.Whatsapp.Dtos.MessageResponse
{
    public class SignalRFlowMessageResponse
    {
        public required string Content { get; set; }
        public required FlowMessageType Type { get; set; }
        public required FlowMessageAction Action { get; set; }
        public required FlowMessageEventType EventType { get; set; }
        public List<SignalRFlowMessageButtonResponse>? Buttons { get; set; }

        public string? ButtonListDisplayText { get; set; }
        public List<SignalRFlowMessageListItemResponse>? ListItems { get; set; }
    }
}
